tx.go

```diff
@@ -85,16 +87,22 @@ func (tx *Tx) Begin() (*Tx, error) {
 func (tx *Tx) RunInTransaction(ctx context.Context, fn func(*Tx) error) error {
 	defer func() {
 		if err := recover(); err != nil {
-			if err := tx.RollbackContext(ctx); err != nil {
-				internal.Logger.Printf(ctx, "tx.Rollback panicked: %s", err)
+			safeCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+			safeCtx = trace.ContextWithRemoteSpanContext(safeCtx, trace.SpanContextFromContext(ctx))
```

Comment:
This should be `trace.ContextWithSpan` since remote span context may have some implications (like with sampling).

But I think I prefer `UndoneContext` since it preserves all values in context...